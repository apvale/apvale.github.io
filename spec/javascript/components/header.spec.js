import { mount, shallowMount } from '@vue/test-utils'
import Header from '@/components/header'
import Menu from '@/components/menu'
import create from '../settings/utils'

describe('Header', () => {
  let props, wrapper

  beforeAll(() => {
    const images = {
      banner: {
        small: 'bar.jpg',
        medium: 'bar.jpg',
        large: 'bar.jpg',
        xlarge: 'bar.jpg',
      }
    }

    window.gon = {
      flag_links: create('menu_items'),
      menu_links: create('menu_items'),
      images: images
    }

    wrapper = mount(Header)
  })

  it('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('has data properties', () => {
    const expected = ['flag_links', 'menu_links', 'images', 'bannerWidth', 'bannerHeight', 'windowWidth', 'windowHeight', 'i18n_css_class', 'main_css_class', 'I18n']
    const received = Object.keys(wrapper.vm.$data)
    expect(received).toEqual(expected)
  })

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })

  describe('Window', () => {
    it('is landscape', () => {
      wrapper.setData({
        windowWidth: 100,
        windowHeight: 90
      })
      expect(wrapper.vm.banner_image_class).toEqual('b-lazy banner-img bgwidth')
    })

    it('is portrait', () => {
      wrapper.setData({
        windowWidth: 90,
        windowHeight: 100
      })
      expect(wrapper.vm.banner_image_class).toEqual('b-lazy banner-img bgheight')
    })
  })

  it.skip('expand_menu', () => {
    wrapper.vm.expand_menu()
  })

  it.skip('destroy', () => {
    wrapper.destroy()
  })

  describe('Child components', () => {
    it('includes Menu', () => {
      const child_component = wrapper.find(Menu)
      expect(child_component.isVueInstance()).toBeTruthy()
    })

    it('renders 2 Menu objects', () => {
      wrapper = shallowMount(Header)
      const menus = wrapper.findAll(Menu)
      expect(menus.length).toBe(2)
    })
  })
})
