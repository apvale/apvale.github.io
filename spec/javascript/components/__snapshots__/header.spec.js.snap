// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<header>
  <div>
    <ul class="i18n">
      <li class="active">
        <a href="#home" class="home">Home</a>
      </li>
      <li>
        <a href="#about" class="about">About</a>
      </li>
    </ul>
  </div>
  <div class="banner">
    <div class="row" style="width: 0px; height: -60px;">
      <div class="col-xs-12 col-sm-12 col-md-12 col-lg-12 no-padd" style="height: 100%;">
        <div class="banner-part">
          <img data-src="bar.jpg" data-src-small="bar.jpg" data-src-medium="bar.jpg" data-src-large="bar.jpg" alt="Banner" class="b-lazy banner-img bgwidth">
        </div>
      </div>
    </div>
    <div class="banner-text container">
      <h1>Gustavo Cunha</h1>
      <h2>Web developer, Mobile and Software Engineer Consultant</h2>
    </div>
  </div>
  <div class="menu">
    <div class="navbar-wrapper is-stuck">
      <div class="container">
        <div class="navwrapper">
          <div class="navbar navbar-inverse navbar-static-top">
            <div class="container">
              <div class="navbar-header">
                <button aria-label="Menu" type="button" data-toggle="collapse" data-target=".navbar-collapse" class="navbar-toggle"><span class="icon-bar"></span> <span class="icon-bar"></span> <span class="icon-bar"></span></button>
              </div>
              <div class="navbar-collapse collapse">
                <ul class="nav navbar-nav">
                  <li class="active">
                    <a href="#home" class="home">Home</a>
                  </li>
                  <li>
                    <a href="#about" class="about">About</a>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</header>
`;
